base/ordering.jl

```diff
@@ -76,4 +85,19 @@ function ord(lt, by, rev::Bool, order::Ordering=Forward)
     return rev ? ReverseOrdering(o) : o
 end
 
+
+# This function is not in use anywhere in Base but we observed
+# use in sorting-related packages (#34719). It's probably best to move
+# this functionality to those packages in the future; let's remind/force
+# ourselves to deprecate this in v2.0.
+# The following clause means `if VERSION < v"2.0-"` but it also works during
+# bootstrap. For the same reason, we need to write `Int32` instead of `Cint`.
```

Comment:
Possibly because `Cint` isn't defined yet? Using `Int32` probably works because of little-endianness: if the return value is actually `Int64` but the caller thinks its `Int32` on a little-endian system, the value is right. Would fail on a big-endian system, although, of course, we don't support any of those.

Reply:
> Possibly because `Cint` isn't defined yet? 

Yes, that's probably it. I didn't attempt to re-shuffle the bootstrap order although it would be nice to have `VERSION` comparisons and safe `ccall`.

> Would fail on a big-endian system, although, of course, we don't support any of those.

Do we support any platforms where `Cint` is different from `Int32` at all? I can only find
```julia
const Cint = Int32
```
in the source. If we do, your analysis about endian-ness is sound.

Reply:
That's a good point, I was thinking of `long` which is 32/64-bit. I guess I haven't been doing much C programming for a while now 😄